boards: shields: Add X-NUCLEO-IDB05A1

erwango opened the pull request:
Add STMicroelectronics BLE shield X-NUCLEO-IDB05A1.
Add configurations for using X-NUCLEO-IDB05A1 with stm32mp157c_dk2 board.

@dbkinder, this PR replaces #18769 since I had to push the PR myself to complete it.
I took most of your suggestions, except the ones I commented

yboutreux committed:
boards: shields: Add x_nucleo_idb05a1

Add shield x_nucleo_idb05a1.
Tested samples with the disco_l475_iot1 (SPI3 disabled to ensure
to test the shield on SPI1, and not the on-board BlueNRG-MS chip):
Beacon, Central and Peripheral have been tested and are working.

Signed-off-by: Yaël Boutreux <yael.boutreux@st.com>
Signed-off-by: Erwan Gouriou <erwang.gouriou@linaro.org>

dbkinder commented on boards/shields/x_nucleo_idb05a1/doc/index.rst:
```diff
@@ -0,0 +1,96 @@
+.. _x-nucleo-idb05a1:
+
+X-NUCLEO-IDB05A1: Bluetooth Low Energy expansion board based on the SPBTLE-RF module
```
I'd (still) reduce the title length to just: ``X-NUCLEO-IDB05A1: Bluetooth Low Energy expansion board`` or maybe even ``X-NUCLEO-IDB05A1: BLE expansion board``

dbkinder commented on boards/shields/x_nucleo_idb05a1/doc/index.rst:
```diff
@@ -0,0 +1,96 @@
+
+You could check Figure 3 in `X-NUCLEO-IDB05A1 databrief`_ for more details.
+
+Known boards that need this IRQ pin modification are :
```
Since this is a list of one, how about:
```
The stm32mp157c_dk2 board is known to need this IRQ pin modification.
```

erwango committed:
shields: x_nucleo_idb05a1: boards: add stm32mp157c_dk2 configuration

Add configuration file for the stm32mp157c_dk2.
Beacon, Central are working.
Peripheral is not compiling because of .elf exceeding FLASH region,
else should be working.

Signed-off-by: Yaël Boutreux <yael.boutreux@st.com>
Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented:
@dbkinder, can you review again? Thanks!

dbkinder reviewed (approved):
thanks.

erwango commented:
@cpriouzeau

MaureenHelm reviewed (changes requested):
Please add a sample or test project configuration to make sure the shield builds in CI. For example:
https://github.com/zephyrproject-rtos/zephyr/blob/ba9b8dd183988ccb3cea6da1d3f9e11cb2ecca14/samples/bluetooth/peripheral_hr/sample.yaml#L9-L13

erwango committed:
samples/bluetooth: peripheral: Support of x_nucleo_idb05a1 ble shield

Aim is to test x_nucleo_idb05a1 shield


Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented:
@MaureenHelm, sample updated, thanks for the proposal !
